Failed: ClusterClientTest

https://travis-ci.org/akka/akka-samples/jobs/616658528?utm_medium=notification&utm_source=github_status

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running sample.cluster.client.grpc.ClusterClientTest
[INFO] [11/25/2019 12:54:47.333] [main] [akka.remote.artery.tcp.ArteryTcpTransport(akka://ClusterClientTest)] Remoting started with transport [Artery tcp]; listening on address [akka://ClusterClientTest@127.0.0.1:39803] with UID [-5395267153817664087]
[INFO] [11/25/2019 12:54:47.368] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Starting up, Akka version [2.6.0-M8] ...
[INFO] [11/25/2019 12:54:47.477] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Registered cluster JMX MBean [akka:type=Cluster]
[INFO] [11/25/2019 12:54:47.477] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Started up successfully
[INFO] [11/25/2019 12:54:47.581] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - No seed-nodes configured, manual cluster join required, see https://doc.akka.io/docs/akka/current/cluster-usage.html#joining-to-seed-nodes
[INFO] [11/25/2019 12:54:47.695] [main] [akka.remote.artery.tcp.ArteryTcpTransport(akka://ClusterClientTest)] Remoting started with transport [Artery tcp]; listening on address [akka://ClusterClientTest@127.0.0.1:39749] with UID [4842462293398171948]
[INFO] [11/25/2019 12:54:47.699] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Starting up, Akka version [2.6.0-M8] ...
[WARN] [11/25/2019 12:54:47.709] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Could not register Cluster JMX MBean with name=akka:type=Cluster as it is already registered. If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config
[INFO] [11/25/2019 12:54:47.710] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Started up successfully
[INFO] [11/25/2019 12:54:47.716] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - No seed-nodes configured, manual cluster join required, see https://doc.akka.io/docs/akka/current/cluster-usage.html#joining-to-seed-nodes
[INFO] [11/25/2019 12:54:47.808] [main] [akka.remote.artery.tcp.ArteryTcpTransport(akka://ClusterClientTest)] Remoting started with transport [Artery tcp]; listening on address [akka://ClusterClientTest@127.0.0.1:42605] with UID [3761732213052592203]
[INFO] [11/25/2019 12:54:47.811] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Starting up, Akka version [2.6.0-M8] ...
[WARN] [11/25/2019 12:54:47.817] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Could not register Cluster JMX MBean with name=akka:type=Cluster as it is already registered. If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config
[INFO] [11/25/2019 12:54:47.817] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Started up successfully
[INFO] [11/25/2019 12:54:47.822] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - No seed-nodes configured, manual cluster join required, see https://doc.akka.io/docs/akka/current/cluster-usage.html#joining-to-seed-nodes
[INFO] [11/25/2019 12:54:47.917] [main] [akka.remote.artery.tcp.ArteryTcpTransport(akka://ClusterClientTest)] Remoting started with transport [Artery tcp]; listening on address [akka://ClusterClientTest@127.0.0.1:37273] with UID [1981176973620231368]
[INFO] [11/25/2019 12:54:47.924] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Starting up, Akka version [2.6.0-M8] ...
[WARN] [11/25/2019 12:54:47.936] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Could not register Cluster JMX MBean with name=akka:type=Cluster as it is already registered. If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config
[INFO] [11/25/2019 12:54:47.937] [main] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Started up successfully
[INFO] [11/25/2019 12:54:47.943] [ClusterClientTest-akka.actor.internal-dispatcher-4] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - No seed-nodes configured, manual cluster join required, see https://doc.akka.io/docs/akka/current/cluster-usage.html#joining-to-seed-nodes
[INFO] [11/25/2019 12:54:48.051] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Node [akka://ClusterClientTest@127.0.0.1:39803] is JOINING itself (with roles [dc-default]) and forming new cluster
[INFO] [11/25/2019 12:54:48.059] [main] [ClusterClientReceptionist] Starting ClusterClientReceptionist gRPC server at 127.0.0.1:50051
[INFO] [11/25/2019 12:54:48.149] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - is the new leader among reachable nodes (more leaders may exist)
[INFO] [11/25/2019 12:54:48.177] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:39803] to [Up]
[INFO] [11/25/2019 12:54:48.926] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Node [akka://ClusterClientTest@127.0.0.1:42605] is JOINING, roles [dc-default]
[INFO] [11/25/2019 12:54:48.935] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Node [akka://ClusterClientTest@127.0.0.1:37273] is JOINING, roles [dc-default]
[INFO] [11/25/2019 12:54:49.020] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Node [akka://ClusterClientTest@127.0.0.1:39749] is JOINING, roles [dc-default]
[INFO] [11/25/2019 12:54:49.029] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Welcome from [akka://ClusterClientTest@127.0.0.1:39803]
[INFO] [11/25/2019 12:54:49.036] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Welcome from [akka://ClusterClientTest@127.0.0.1:39803]
[INFO] [11/25/2019 12:54:49.048] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Welcome from [akka://ClusterClientTest@127.0.0.1:39803]
[INFO] [11/25/2019 12:54:49.615] [main] [ClusterClientReceptionist] Starting ClusterClientReceptionist gRPC server at 127.0.0.1:50052
[INFO] [11/25/2019 12:54:49.625] [Thread-9] [ClusterClientReceptionist] ClusterClientReceptionist gRPC server stopped successfully
[INFO] [11/25/2019 12:54:49.628] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:37273] to [Up]
[INFO] [11/25/2019 12:54:49.628] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:39749] to [Up]
[INFO] [11/25/2019 12:54:49.628] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:42605] to [Up]
[INFO] [11/25/2019 12:54:49.660] [main] [ClusterClientReceptionist] Starting ClusterClientReceptionist gRPC server at 127.0.0.1:50053
[INFO] [11/25/2019 12:54:49.664] [Thread-10] [ClusterClientReceptionist] ClusterClientReceptionist gRPC server stopped successfully
[INFO] [11/25/2019 12:54:49.684] [main] [ClusterClientReceptionist] Starting ClusterClientReceptionist gRPC server at 127.0.0.1:50054
[INFO] [11/25/2019 12:54:49.697] [Thread-11] [ClusterClientReceptionist] ClusterClientReceptionist gRPC server stopped successfully
[INFO] [11/25/2019 12:54:49.734] [Thread-12] [ClusterClientReceptionist] ClusterClientReceptionist gRPC server stopped successfully
[INFO] [11/25/2019 12:54:50.629] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - is no longer leader
[INFO] [11/25/2019 12:54:50.998] [ClusterClientTest-akka.actor.internal-dispatcher-4] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - is the new leader among reachable nodes (more leaders may exist)
[INFO] [11/25/2019 12:54:52.779] [ClusterClientTest-akka.actor.default-dispatcher-6] [akka://ClusterClientTest/user/client1] New session for Actor[akka://ClusterClientTest/system/testProbe-1#-1772270167]
[INFO] [11/25/2019 12:54:53.926] [ClusterClientTest-akka.actor.default-dispatcher-8] [ClusterClientReceptionist] New session [0a151923-45a7-4163-a49b-631f75a85ed0]
[INFO] [11/25/2019 12:54:54.242] [ClusterClientTest-akka.actor.default-dispatcher-9] [akka://ClusterClientTest/user/ask-sesion-client] New session for Actor[akka://ClusterClientTest/temp/$a]
[INFO] [11/25/2019 12:54:54.278] [ClusterClientTest-akka.actor.default-dispatcher-8] [ClusterClientReceptionist] New session [30fc2d3b-a763-4785-849a-63267c255fcc]
[INFO] [11/25/2019 12:54:54.279] [Thread-13] [ClusterClientReceptionist] Session [0a151923-45a7-4163-a49b-631f75a85ed0] completed successfully: Done
[INFO] [11/25/2019 12:54:54.305] [ClusterClientTest-akka.actor.default-dispatcher-9] [akka://ClusterClientTest/user/ask-sesion-client] New session for Actor[akka://ClusterClientTest/temp/$b]
[INFO] [11/25/2019 12:54:54.319] [Thread-14] [akka://ClusterClientTest/user/ask-sesion-client] Session completed successfully for Actor[akka://ClusterClientTest/temp/$a]: Done
[INFO] [11/25/2019 12:54:54.338] [ClusterClientTest-akka.actor.default-dispatcher-8] [ClusterClientReceptionist] New session [0d8de4b7-8743-41e5-9e9e-1e8d064bccda]
[INFO] [11/25/2019 12:54:54.341] [Thread-15] [ClusterClientReceptionist] Session [30fc2d3b-a763-4785-849a-63267c255fcc] completed with failure: akka.http.scaladsl.model.http2.PeerClosedStreamException: Stream with ID [3] was closed by peer with code CANCEL(0x08)
[INFO] [11/25/2019 12:54:54.371] [Thread-16] [akka://ClusterClientTest/user/ask-sesion-client] Session completed successfully for Actor[akka://ClusterClientTest/temp/$b]: Done
[INFO] [11/25/2019 12:54:54.375] [Thread-17] [ClusterClientReceptionist] Session [0d8de4b7-8743-41e5-9e9e-1e8d064bccda] completed with failure: akka.http.scaladsl.model.http2.PeerClosedStreamException: Stream with ID [5] was closed by peer with code CANCEL(0x08)
[INFO] [11/25/2019 12:54:55.795] [ClusterClientTest-akka.actor.default-dispatcher-11] [akka://ClusterClientTest/user/client] New session for Actor[akka://ClusterClientTest/system/testProbe-1#-1772270167]
[INFO] [11/25/2019 12:54:55.818] [ClusterClientTest-akka.actor.default-dispatcher-8] [ClusterClientReceptionist] New session [b699b262-c662-4ac4-9b62-4a4e1f37b3a0]
[INFO] [11/25/2019 12:54:55.841] [Thread-24] [ClusterClientReceptionist] Session [b699b262-c662-4ac4-9b62-4a4e1f37b3a0] completed successfully: Done
[INFO] [11/25/2019 12:54:55.930] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Marked address [akka://ClusterClientTest@127.0.0.1:39803] as [Leaving]
[INFO] [11/25/2019 12:54:57.138] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:39803] to [Exiting]
[INFO] [11/25/2019 12:54:57.147] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Exiting completed
[INFO] [11/25/2019 12:54:57.151] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Shutting down...
[INFO] [11/25/2019 12:54:57.151] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39803] - Successfully shut down
[INFO] [11/25/2019 12:54:57.163] [ClusterClientTest-akka.actor.internal-dispatcher-4] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Exiting confirmed [akka://ClusterClientTest@127.0.0.1:39803]
[INFO] [11/25/2019 12:54:57.169] [ClusterClientTest-akka.actor.default-dispatcher-28] [akka://ClusterClientTest/system/cluster/core/daemon] Message [akka.cluster.GossipEnvelope] from Actor[akka://ClusterClientTest@127.0.0.1:42605/system/cluster/core/daemon#2081416629] to Actor[akka://ClusterClientTest/system/cluster/core/daemon#1621349979] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://ClusterClientTest/system/cluster/core/daemon#1621349979] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [11/25/2019 12:54:57.171] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka://ClusterClientTest@127.0.0.1:39803/system/remoting-terminator] Shutting down remote daemon.
[INFO] [11/25/2019 12:54:57.174] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka://ClusterClientTest@127.0.0.1:39803/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [11/25/2019 12:54:57.174] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Exiting confirmed [akka://ClusterClientTest@127.0.0.1:39803]
[WARN] [11/25/2019 12:54:57.213] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-2)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:42605], message stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:54:57.216] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-2)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:39749], control stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:54:57.219] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-2)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:39749], message stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:54:57.221] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-2)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:37273], message stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:54:57.222] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-2)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:37273], control stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:54:57.224] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-2)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:42605], control stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[INFO] [11/25/2019 12:54:57.249] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka://ClusterClientTest@127.0.0.1:39803/system/remoting-terminator] Remoting shut down.
[INFO] [11/25/2019 12:54:57.282] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Marked address [akka://ClusterClientTest@127.0.0.1:39749] as [Leaving]
[INFO] [11/25/2019 12:54:58.202] [ClusterClientTest-akka.remote.default-remote-dispatcher-12] [akka.remote.artery.Association(akka://ClusterClientTest)] Association to [akka://ClusterClientTest@127.0.0.1:39803] having UID [-5395267153817664087] has been stopped. All messages to this UID will be delivered to dead letters. Reason: ActorSystem terminated
[INFO] [11/25/2019 12:54:58.207] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.remote.artery.Association(akka://ClusterClientTest)] Association to [akka://ClusterClientTest@127.0.0.1:39803] having UID [-5395267153817664087] has been stopped. All messages to this UID will be delivered to dead letters. Reason: ActorSystem terminated
[INFO] [11/25/2019 12:54:58.207] [ClusterClientTest-akka.remote.default-remote-dispatcher-13] [akka.remote.artery.Association(akka://ClusterClientTest)] Association to [akka://ClusterClientTest@127.0.0.1:39803] having UID [-5395267153817664087] has been stopped. All messages to this UID will be delivered to dead letters. Reason: ActorSystem terminated
[INFO] [11/25/2019 12:54:59.189] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:39749] to [Exiting]
[INFO] [11/25/2019 12:54:59.190] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Leader is removing confirmed Exiting node [akka://ClusterClientTest@127.0.0.1:39803]
[INFO] [11/25/2019 12:54:59.219] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Exiting completed
[INFO] [11/25/2019 12:54:59.226] [ClusterClientTest-akka.actor.internal-dispatcher-4] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Exiting confirmed [akka://ClusterClientTest@127.0.0.1:39749]
[INFO] [11/25/2019 12:54:59.240] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Shutting down...
[INFO] [11/25/2019 12:54:59.242] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Exiting confirmed [akka://ClusterClientTest@127.0.0.1:39749]
[WARN] [11/25/2019 12:54:59.246] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Could not unregister Cluster JMX MBean with name=akka:type=Cluster as it was not found. If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config
[INFO] [11/25/2019 12:54:59.250] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:39749] - Successfully shut down
[INFO] [11/25/2019 12:54:59.252] [ClusterClientTest-akka.actor.default-dispatcher-10] [akka://ClusterClientTest/system/cluster/core/daemon] Message [akka.cluster.GossipEnvelope] from Actor[akka://ClusterClientTest@127.0.0.1:37273/system/cluster/core/daemon#1628629284] to Actor[akka://ClusterClientTest/system/cluster/core/daemon#509197998] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://ClusterClientTest/system/cluster/core/daemon#509197998] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [11/25/2019 12:54:59.255] [ClusterClientTest-akka.actor.default-dispatcher-8] [akka://ClusterClientTest/system/cluster/core/daemon] Message [akka.cluster.GossipEnvelope] from Actor[akka://ClusterClientTest@127.0.0.1:37273/system/cluster/core/daemon#1628629284] to Actor[akka://ClusterClientTest/system/cluster/core/daemon#509197998] was not delivered. [2] dead letters encountered. If this is not an expected behavior then Actor[akka://ClusterClientTest/system/cluster/core/daemon#509197998] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [11/25/2019 12:54:59.259] [ClusterClientTest-akka.actor.default-dispatcher-8] [akka://ClusterClientTest/system/cluster/core/daemon] Message [akka.cluster.GossipEnvelope] from Actor[akka://ClusterClientTest@127.0.0.1:42605/system/cluster/core/daemon#2081416629] to Actor[akka://ClusterClientTest/system/cluster/core/daemon#509197998] was not delivered. [3] dead letters encountered. If this is not an expected behavior then Actor[akka://ClusterClientTest/system/cluster/core/daemon#509197998] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [11/25/2019 12:54:59.260] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka://ClusterClientTest@127.0.0.1:39749/system/remoting-terminator] Shutting down remote daemon.
[INFO] [11/25/2019 12:54:59.261] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka://ClusterClientTest@127.0.0.1:39749/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [11/25/2019 12:55:00.208] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Leader is removing confirmed Exiting node [akka://ClusterClientTest@127.0.0.1:39749]
[INFO] [11/25/2019 12:55:00.277] [ClusterClientTest-akka.remote.default-remote-dispatcher-13] [akka.remote.artery.Association(akka://ClusterClientTest)] Association to [akka://ClusterClientTest@127.0.0.1:39749] having UID [4842462293398171948] has been stopped. All messages to this UID will be delivered to dead letters. Reason: ActorSystem terminated
[WARN] [11/25/2019 12:55:00.283] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-5)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:37273], control stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:55:00.284] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-5)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:37273], message stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:55:00.284] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-5)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:42605], control stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:55:00.284] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-5)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:42605], message stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[INFO] [11/25/2019 12:55:00.285] [ClusterClientTest-akka.remote.default-remote-dispatcher-12] [akka.remote.artery.Association(akka://ClusterClientTest)] Association to [akka://ClusterClientTest@127.0.0.1:39749] having UID [4842462293398171948] has been stopped. All messages to this UID will be delivered to dead letters. Reason: ActorSystem terminated
[INFO] [11/25/2019 12:55:00.293] [ClusterClientTest-akka.remote.default-remote-dispatcher-11] [akka://ClusterClientTest@127.0.0.1:39749/system/remoting-terminator] Remoting shut down.
[INFO] [11/25/2019 12:55:00.315] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Marked address [akka://ClusterClientTest@127.0.0.1:42605] as [Leaving]
[INFO] [11/25/2019 12:55:01.230] [ClusterClientTest-akka.actor.internal-dispatcher-4] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:42605] to [Exiting]
[INFO] [11/25/2019 12:55:01.236] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Exiting completed
[INFO] [11/25/2019 12:55:01.237] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Shutting down...
[WARN] [11/25/2019 12:55:01.238] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Could not unregister Cluster JMX MBean with name=akka:type=Cluster as it was not found. If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config
[INFO] [11/25/2019 12:55:01.238] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:42605] - Successfully shut down
[INFO] [11/25/2019 12:55:01.251] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Exiting confirmed [akka://ClusterClientTest@127.0.0.1:42605]
[INFO] [11/25/2019 12:55:01.252] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka://ClusterClientTest@127.0.0.1:42605/system/remoting-terminator] Shutting down remote daemon.
[INFO] [11/25/2019 12:55:01.252] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka://ClusterClientTest@127.0.0.1:42605/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[ERROR] [11/25/2019 12:55:02.066] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka.remote.artery.Association(akka://ClusterClientTest)] Outbound message stream to [akka://ClusterClientTest@127.0.0.1:39803] failed after shutdown.  (akka.remote.artery.Association$OutboundStreamStopQuarantinedSignal$: )
[ERROR] [11/25/2019 12:55:02.066] [ClusterClientTest-akka.remote.default-remote-dispatcher-12] [akka.remote.artery.Association(akka://ClusterClientTest)] Outbound control stream to [akka://ClusterClientTest@127.0.0.1:39803] failed after shutdown.  (akka.remote.artery.Association$OutboundStreamStopQuarantinedSignal$: )
[INFO] [11/25/2019 12:55:02.207] [ClusterClientTest-akka.actor.default-dispatcher-9] [akka://ClusterClientTest/deadLetters] Message [akka.remote.artery.SystemMessageDelivery$ClearSystemMessageDelivery] to Actor[akka://ClusterClientTest/deadLetters] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://ClusterClientTest/deadLetters] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [11/25/2019 12:55:02.249] [ClusterClientTest-akka.actor.internal-dispatcher-5] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Leader is removing confirmed Exiting node [akka://ClusterClientTest@127.0.0.1:42605]
[INFO] [11/25/2019 12:55:02.267] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka.remote.artery.Association(akka://ClusterClientTest)] Association to [akka://ClusterClientTest@127.0.0.1:42605] having UID [3761732213052592203] has been stopped. All messages to this UID will be delivered to dead letters. Reason: ActorSystem terminated
[WARN] [11/25/2019 12:55:02.276] [ClusterClientTest-akka.remote.default-remote-dispatcher-12] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-8)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:37273], control stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[WARN] [11/25/2019 12:55:02.276] [ClusterClientTest-akka.remote.default-remote-dispatcher-7] [akka.stream.Log(akka://ClusterClientTest/system/Materializers/StreamSupervisor-8)] [outbound connection to [akka://ClusterClientTest@127.0.0.1:37273], message stream] Upstream failed, cause: StreamTcpException: The connection has been aborted
[INFO] [11/25/2019 12:55:02.280] [ClusterClientTest-akka.remote.default-remote-dispatcher-12] [akka://ClusterClientTest@127.0.0.1:42605/system/remoting-terminator] Remoting shut down.
[INFO] [11/25/2019 12:55:02.298] [ClusterClientTest-akka.actor.internal-dispatcher-3] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Marked address [akka://ClusterClientTest@127.0.0.1:37273] as [Leaving]
[INFO] [11/25/2019 12:55:03.267] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Leader is moving node [akka://ClusterClientTest@127.0.0.1:37273] to [Exiting]
[INFO] [11/25/2019 12:55:03.272] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Exiting completed
[INFO] [11/25/2019 12:55:03.273] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Shutting down...
[WARN] [11/25/2019 12:55:03.274] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Could not unregister Cluster JMX MBean with name=akka:type=Cluster as it was not found. If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config
[INFO] [11/25/2019 12:55:03.275] [ClusterClientTest-akka.actor.internal-dispatcher-2] [akka.cluster.Cluster(akka://ClusterClientTest)] Cluster Node [akka://ClusterClientTest@127.0.0.1:37273] - Successfully shut down
[INFO] [11/25/2019 12:55:03.281] [ClusterClientTest-akka.remote.default-remote-dispatcher-13] [akka://ClusterClientTest@127.0.0.1:37273/system/remoting-terminator] Shutting down remote daemon.
[INFO] [11/25/2019 12:55:03.282] [ClusterClientTest-akka.remote.default-remote-dispatcher-13] [akka://ClusterClientTest@127.0.0.1:37273/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[INFO] [11/25/2019 12:55:03.289] [ClusterClientTest-akka.remote.default-remote-dispatcher-13] [akka://ClusterClientTest@127.0.0.1:37273/system/remoting-terminator] Remoting shut down.
Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 18.486 sec <<< FAILURE!
step1_shouldJoinNodesToCluster(sample.cluster.client.grpc.ClusterClientTest)  Time elapsed: 4.673 sec  <<< FAILURE!
java.lang.AssertionError: assertion failed: expected 1, found 0
	at scala.Predef$.assert(Predef.scala:223)
	at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:419)
	at akka.testkit.TestKitBase.expectMsg(TestKit.scala:395)
	at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:395)
	at akka.testkit.TestKit.expectMsg(TestKit.scala:928)
	at akka.testkit.javadsl.TestKit.expectMsg(TestKit.scala:464)
	at sample.cluster.client.grpc.ClusterClientTest.lambda$awaitCount$0(ClusterClientTest.java:147)
	at akka.testkit.javadsl.TestKit.$anonfun$awaitAssert$1(TestKit.scala:379)
	at akka.testkit.TestKitBase.poll$2(TestKit.scala:327)
	at akka.testkit.TestKitBase.awaitAssert(TestKit.scala:344)
	at akka.testkit.TestKitBase.awaitAssert$(TestKit.scala:316)
	at akka.testkit.TestKit.awaitAssert(TestKit.scala:928)
	at akka.testkit.javadsl.TestKit.awaitAssert(TestKit.scala:379)
	at sample.cluster.client.grpc.ClusterClientTest.awaitCount(ClusterClientTest.java:144)
	at sample.cluster.client.grpc.ClusterClientTest.step1_shouldJoinNodesToCluster(ClusterClientTest.java:170)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Results :
Failed tests:   step1_shouldJoinNodesToCluster(sample.cluster.client.grpc.ClusterClientTest): assertion failed: expected 1, found 0
```

--- a/akka-sample-cluster-client-grpc-java/src/test/java/sample/cluster/client/grpc/ClusterClientTest.java
+++ b/akka-sample-cluster-client-grpc-java/src/test/java/sample/cluster/client/grpc/ClusterClientTest.java
@@ -141,7 +141,7 @@ public class ClusterClientTest {
 
   private void awaitCount(ActorSystem system, int expected, Duration timeout) {
     TestKit testKit = new TestKit(system);
-    testKit.awaitAssert(() -> {
+    testKit.awaitAssert(timeout, () -> {
       // FIXME Java API missing for accessing Count instance https://github.com/akka/akka/issues/27728
       DistributedPubSub.get(system).mediator().tell(DistributedPubSubMediator.Count$.MODULE$, testKit.getRef());
       testKit.expectMsg(expected);

